# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditor.h
# Commit: a43e95a534eb
# Full Hash: a43e95a534eb5c13883cc2a61fb64c15fe29e674
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-10-05 03:40:21
# Regressor Bug: 1792654
# File Overlap Count: 1
# Description:
#   Bug 1792654 - Make `CreateNodeResultBase`, `MoveNodeResult`, `SplitNodeResult` and `SplitRangeOffResult` handle caret point with a common base class r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D158335
# ==============================================================================

diff -r 18196c5fab62 -r a43e95a534eb editor/libeditor/HTMLEditor.h
--- a/editor/libeditor/HTMLEditor.h	Tue Oct 04 21:13:53 2022 +0000
+++ b/editor/libeditor/HTMLEditor.h	Tue Oct 04 21:36:57 2022 +0000
@@ -4632,20 +4632,16 @@
                                                // CollectNonEditableNodes
   friend class ListItemElementSelectionState;  // CollectEditTargetNodes,
                                                // CollectNonEditableNodes
-  friend class MoveNodeResult;       // AllowsTransactionsToChangeSelection,
-                                     // CollapseSelectionTo
   friend class MoveNodeTransaction;  // AllowsTransactionsToChangeSelection,
                                      // CollapseSelectionTo, MarkElementDirty,
                                      // RangeUpdaterRef
-  friend class ParagraphStateAtSelection;    // CollectChildren,
-                                             // CollectEditTargetNodes,
-                                             // CollectListChildren,
-                                             // CollectNonEditableNodes,
-                                             // CollectTableChildren
-  friend class SlurpBlobEventListener;       // BlobReader
-  friend class SplitNodeResult;              // CollapseSelectionTo
-  friend class SplitNodeTransaction;         // DoJoinNodes, DoSplitNode
-  friend class SplitRangeOffFromNodeResult;  // CollapseSelectionTo
+  friend class ParagraphStateAtSelection;  // CollectChildren,
+                                           // CollectEditTargetNodes,
+                                           // CollectListChildren,
+                                           // CollectNonEditableNodes,
+                                           // CollectTableChildren
+  friend class SlurpBlobEventListener;     // BlobReader
+  friend class SplitNodeTransaction;       // DoJoinNodes, DoSplitNode
   friend class TransactionManager;  // DidDoTransaction, DidRedoTransaction,
                                     // DidUndoTransaction
   friend class
